lib/src/query.c

```diff
@@ -2323,6 +2325,8 @@ static TSQueryError ts_query__parse_pattern(
         // Parse the wildcard symbol
         if (length == 1 && node_name[0] == '_') {
           symbol = WILDCARD_SYMBOL;
+        } else if (!strncmp(node_name, "MISSING", length)) {
+          symbol = MISSING_SYMBOL;
```

Comment:
instead of "MISSING_SYMBOL" (this is semantically *incorrect*), just add a bool "is_missing" with symbol.